diag_manager/fms_diag_output_buffer.F90

```diff
@@ -494,17 +551,17 @@ subroutine allocate_buffer_3d(this, buff_type, buff_sizes, field_name, diurnal_s
   select type (buff_type)
     type is (integer(kind=i4_kind))
       allocate(integer(kind=i4_kind) :: this%buffer( buff_sizes(1),buff_sizes(2), buff_sizes(3)))
-      allocate(integer(kind=i4_kind) :: this%counter(buff_sizes(1),buff_sizes(2), buff_sizes(3)))
-      allocate(integer(kind=i4_kind) :: this%count_0d(n_samples))
-      this%counter = 0_i4_kind
-      this%count_0d = 0_i4_kind
+      allocate(real(kind=r4_kind) :: this%counter(buff_sizes(1),buff_sizes(2), buff_sizes(3)))
```

Comment:
Right.  That should be documented in the code where they are allocated.

Reply:
I don't understand the math here:

When averaging, the buffer is initialized to 0 and each time period the data is multiplied by the weight (which will be 1 if it not passed in the send_data call) and added to the buffer

https://github.com/NOAA-GFDL/FMS/blob/dd53551c20345bd902786ec84a0e37a1820b0df3/diag_manager/diag_manager.F90#L2046-L2048

The counter is initialized to 0 and the weight gets added after every time
https://github.com/NOAA-GFDL/FMS/blob/dd53551c20345bd902786ec84a0e37a1820b0df3/diag_manager/diag_manager.F90#L2050-L2051

When it is time to write, it divides the buffer by the counter and that is the "average" that gets written:
https://github.com/NOAA-GFDL/FMS/blob/dd53551c20345bd902786ec84a0e37a1820b0df3/diag_manager/diag_manager.F90#L3466-L3467

I don't understand why the weight gets added to the counter, shouldn't it just add 1 every time? 

I **think** that the reason why it is an array and not a single value is because when sending in blocks of data only a portion of the buffer gets updated so the counter needs to be tracked in a per grid point basis.

Reply:
I think the weight doesn't have to be 1 necessarily, 1.0 is the default but otherwise any real value can get passed in to send data:

https://github.com/NOAA-GFDL/FMS/blob/07ff0679281f4d99423483f33eb7fc4240f5dc88/diag_manager/diag_manager.F90#L1882-L1895

Reply:
@uramirez8707 @rem1776 is this conversation resolved?